clang/include/clang/Parse/Parser.h

```diff
@@ -7213,7 +7213,8 @@ class Parser : public CodeCompletionHandler {
   /// 'while', or 'for').
   StmtResult
   ParseStatement(SourceLocation *TrailingElseLoc = nullptr,
-                 ParsedStmtContext StmtCtx = ParsedStmtContext::SubStmt);
+                 ParsedStmtContext StmtCtx = ParsedStmtContext::SubStmt,
+                 SmallVectorImpl<LabelDecl *> *LoopOrSwitchNames = nullptr);
```

Comment:
Given conversations with the paper author and feedback we've given on it, do we want to track only a single name rather than lists of names?

That isn't a huge win in terms of simplicity with these interfaces, but it does make downstream consumers of the AST easier to reason about (think: AST matchers that want to ask 'what's the name of this loop?' and that kind of thing).

Reply:
Done; that simplified everything a bit.